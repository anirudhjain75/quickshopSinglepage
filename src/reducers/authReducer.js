const initialState = {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    login: false
};

export default (state = initialState, action) => {
    switch(action.type)
    {
        case 'EMAIL_CHANGED':
            return {
                ...state,
                email: action.payload
            };
        case 'PASSWORD_CHANGED':
            return {
                ...state,
                password: action.payload
            };
        case 'FIRST_NAME_CHANGED':
            return {
                ...state,
                firstName: action.payload
            };
        case 'LAST_NAME_CHANGED':
            return {
                ...state,
                lastName: action.payload
            };
        default:
            return state;
    }
}