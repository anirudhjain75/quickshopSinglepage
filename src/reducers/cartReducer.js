const initialState = {
    cart: []
};

export default (state = initialState, action) => {
    switch (action.type) {
        case 'ADD_TO_CART' :
            return {
                ...state,
                cart: [ ...state.cart, action.payload]
            };
        case 'REMOVE_FROM_CART' : {
            let temp = state.cart;
            const index = temp.indexOf(action.payload);
            temp.splice(index, 1);
            return {
                ...state,
                cart: temp
            };
        }
        default:
            return state;
    }
}