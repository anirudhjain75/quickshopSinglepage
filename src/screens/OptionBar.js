import React from 'react';
import { View, Text, TouchableOpacity, Button } from 'react-native';
import { connect } from 'react-redux';

import {logOut} from "../actions";

class OptionBar extends React.Component {

    showLoginOptions() {
        if(!this.props.login) {
            return (
                <View style={{marginTop: 30, flexDirection: 'row', justifyContent: 'space-evenly', borderBottomWidth: 1, borderBottomColor: '#c3c3c3', paddingBottom: 20}}>
                    <TouchableOpacity
                        style={{height: 25, width: 100, backgroundColor: 'orange', alignItems: 'center', justifyContent: 'center'}}
                        onPress={() => {
                            this.props.navigator.toggleDrawer({
                                to: 'closed',
                                side: 'left',
                                animated: 'true'
                            });
                            this.props.navigator.showModal({screen: 'quickshop.loginform', title: 'Login', animation: 'none'})
                        }}
                    >
                        <Text style={{fontSize: 14, color: 'white'}}> Login </Text>
                    </TouchableOpacity>
                    <TouchableOpacity
                        style={{height: 25, width: 100, backgroundColor: 'orange', alignItems: 'center', justifyContent: 'center'}}
                        onPress={() => {
                            this.props.navigator.toggleDrawer({
                                to: 'closed',
                                side: 'left',
                                animated: 'true'
                            });
                            this.props.navigator.showModal({screen: 'quickshop.signup', title: 'Sign Up', animation: 'none'})
                        }}
                    >
                        <Text style={{fontSize: 14, color: 'white'}}> Sign Up </Text>
                    </TouchableOpacity>
                </View>
            )
        }
        return null
    }

    showLogout() {
        if (this.props.login) {
            return <Button
                title={'Logout'}
                style={{backgroundColor: 'red'}}
                onPress={() => {
                    this.props.navigator.toggleDrawer({
                        to: 'closed',
                        side: 'left',
                        animated: 'true'
                    });
                    this.props.logOut();
                }}
            />
        }
    }

    render() {
        return (
            <View style={{paddingLeft: 5}}>
                <View style={{marginTop: 40, marginLeft: 40}}>
                    <Text style={{fontSize: 16}}>{`Hi ` + (this.props.name || 'Guest')}</Text>
                </View>
                {this.showLoginOptions()}
                <View style={{marginTop: 15}}>
                    <Button
                        title={'Home'}
                        onPress={() => {
                                this.props.navigator.pop({
                                    animated: true,
                                    animationType: 'fade'
                                })}
                        }
                    />
                    <Button
                        title={'Popular Products'}
                        onPress={() => null}
                    />
                </View>
                {this.showLogout()}
            </View>
        )
    }
}

const mapStateToProps = state => {
  return { name: state.auth.firstName, login: state.auth.login}
};

export default connect(mapStateToProps, {logOut})(OptionBar);