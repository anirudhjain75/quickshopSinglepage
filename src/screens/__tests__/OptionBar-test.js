import React from 'react';
import OptionBar from '../OptionBar';
import {Provider} from 'react-redux';

import renderer from 'react-test-renderer';
import reducers from "../../reducers";
import {createStore} from "redux";

const state = createStore(reducers, {});

test("Option Bar renders properly", () => {
    const options = renderer.create(<Provider store={state}><OptionBar/></Provider>).toJSON();
    expect(options).toMatchSnapshot();
})