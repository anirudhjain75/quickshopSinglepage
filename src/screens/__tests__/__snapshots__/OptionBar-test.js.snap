// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Bar renders properly 1`] = `
<View
  style={
    Object {
      "paddingLeft": 5,
    }
  }
>
  <View
    style={
      Object {
        "marginLeft": 40,
        "marginTop": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Hi Guest
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#c3c3c3",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginTop": 30,
        "paddingBottom": 20,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "orange",
          "height": 25,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
          }
        }
      >
         Login 
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "orange",
          "height": 25,
          "justifyContent": "center",
          "opacity": 1,
          "width": 100,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
          }
        }
      >
         Sign Up 
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 15,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Home
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Popular Products
        </Text>
      </View>
    </View>
  </View>
</View>
`;
